src/amber/cli/generators/generator.cr

```diff
@@ -0,0 +1,81 @@
+require "../helpers/migration"
+require "./field.cr"
+
+module Amber::CLI
+  abstract class Generator < Teeplate::FileTree
+    include Helpers::Migration
+    include Helpers
+
+    property name : String
+    property fields : Array(Field)
+    property fields_hash : Hash(String, String)
+    property config : Amber::CLI::Config
+    property table_name : String?
+    property timestamp : String
+
+    def initialize(@name, params)
+      @config = CLI.config
+      @table_name ||= name_plural
+      @fields = parse_fields(params)
+      @fields_hash = parse_fields_hash
+      @timestamp = Time.now.to_s("%Y%m%d%H%M%S%L")
+    end
+
+    def filter(entries)
+      entries.reject { |entry| entry.path.includes?("src/views") && !entry.path.includes?("#{config.language}") }
+    end
+
+    def render(directory)
+      pre_render(directory)
+      super(directory, list: true, color: true)
+      post_render(directory)
+    end
+
+    def pre_render(directory)
+    end
+
+    def post_render(directory)
+    end
+
+    def add_timestamp_fields
+      @fields += %w(created_at:time updated_at:time).map do |f|
+        Field.new(f, hidden: true, database: config.database)
+      end
+    end
+
+    private def parse_fields(params : Array(String)?)
+      if params
+        fields = params.map { |field| Field.new(field, database: config.database) }
+      else
+        fields = [] of Field
+      end
+      return fields
+    end
+
+    private def parse_fields_hash
+      fields_hash = Hash(String, String).new
+      @fields.reject(&.hidden).each do |f|
+        field_name = f.reference? ? "#{f.name}_id" : f.name
+        fields_hash[field_name] = default_value(f.cr_type) unless f.nil?
+      end
+      fields_hash
+    end
+
+    private def default_value(field_type)
```

Comment:
Should default values be give as a definition param instead of assumed by the generator?

Reply:
That would be a nice feature but not sure how you would do that.  This is used by the specs.  It would be cool if we had a `Faker` shard instead.